fava/plugins/documents_from_metadata.py

```diff
@@ -0,0 +1,87 @@
+"""Plugin that looks through all transactions and their postings for the
+`statement`-metadata-key, and if found and present in the filesystem, adds a
+corresponding Document-directive.
+
+The file specified in the `statement`-metadata-value will be searched
+"intelligently", based on the suggestions by @corani
+(https://github.com/beancount/fava/issues/386#issuecomment-256267212).
+"""
+import collections
+from os.path import join, dirname, isfile
+from beancount.core import data
+from beancount.core.compare import hash_entry
+
+StatementDocumentError = collections.namedtuple('StatementDocumentError',
+                                                'source message entry')
+
+__plugins__ = ['statements_from_metadata']
+
+
+def statements_from_metadata(entries, options_map):
+    errors = []
+
+    if 'documents' not in options_map or len(options_map['documents']) == 0:
+        return entries, errors
+
+    def filenames(date, name):
+        d_str = date.strftime("%Y-%m-%d")
+        return [name, '{} {}'.format(d_str, name), '{}.{}'.format(d_str, name)]
+
+    for i, entry in enumerate(entries):
+        type = entry.__class__.__name__.lower()
+
+        if type != 'transaction':
+            continue
+
+        paths = []  # set of tuples (account, path)
+        b_dir = dirname(options_map['filename'])
+
+        for documents in options_map['documents']:
+            statements = []
+
+            # If the `statement` metadata key is set on the transaction, assume
+            # it belongs to the first posting
```

Comment:
This is surprising behaviour to me. The "fuzzy" part should cover this and find the right posting IMHO.